FreydCategoriesForCAP/gap/RingsAsAbCats.gi

```diff
@@ -372,6 +372,29 @@ InstallGlobalFunction( INSTALL_FUNCTIONS_FOR_RING_AS_CATEGORY,
         
     fi;
     
+    
+    ## Random Methods
+    
+    AddRandomObjectByInteger( category,
+      { cat, n } -> RingAsCategoryUniqueObject( cat ) );
+    
+    AddRandomMorphismWithFixedSourceAndRangeByInteger( category,
+      { cat, S, R, n } ->  RingAsCategoryMorphism( cat, Sum( [ 1 .. n ], i -> Random( ring ), Zero( ring ) ) ) );
+    
+    AddRandomObjectByList( category,
+      { cat, L } -> RingAsCategoryUniqueObject( cat ) );
+    
+    AddRandomMorphismWithFixedSourceAndRangeByList( category,
+      function( cat, S, R, L )
+        
+        if Length( L ) <> 1 or not IsInt( L[1] ) then
+            Error( "the input should be a list consisting of one integer!\n" );
+        fi;
+        
+        return RandomMorphismWithFixedSourceAndRangeByInteger( cat, S, R, L[1] );
+        
+    end );
```

Comment:
Do we really need this? If yes, maybe this could be a final derivation mapping all the "ByList" variants to the "ByInteger" variants by requiring an integers as the single entry.

Reply:
No, not really, I only added it for the sake of completeness and for testing the methods I added in the additive closure category.  Anyway, the bundle idea is a good.  I will implement it.